src/services/infrastructure/AvatarGenerator.ts

```diff
@@ -30,15 +31,34 @@ class AvatarGenerator {
     return randomAvatar;
   }
 
-  generateAvatarList(numberOfAvatars: number): string[] {
+  nameToColour(name: string) {
+    const stringArray = Array.from(name);
+    let stringUniqueHash = [...stringArray].reduce((acc, char) => {
+      return char.charCodeAt(0) + ((acc << 5) - acc);
+    }, 0);
+    return `hsl(${stringUniqueHash % 360}, 95%, 35%, 0.5)`;
+  }
+
+  generateAvatarList(members: TeamMember[]): Avatar[] {
     let avatarArray = [];
-    for (let i = 0; i < numberOfAvatars; i++) {
-      avatarArray.push(this.randomise());
+    for (let i = 0; i < members.length; i++) {
```

Comment:
you can use a map here instead of a for loop.

```js
const avatarArray = members.map((member)=> {
       const link = this.randomise();
       const bgColor = this.nameToColour(member.fullName);
       return  {
         link,
         bgColor,
       };
})
```